tests/datatypes/test_str.py

```diff
@@ -574,6 +574,23 @@ def test_rjust(self):
                 print(err)
             """)
 
+    def test_rpartition(self):
+        self.assertCodeExecution("""
+            st = "Hello World!"
+            print(st.rpartition("H"))
+            print(st.rpartition(" "))
+            print(st.rpartition("l"))
+            print(st.rpartition("q"))
+            print(st.rpartition("lo"))
+            print(st.rpartition("!"))
+            print(st.rpartition("ld!"))
+            print("voc".rpartition("foobar"))
+            try:
+                print(st.rpartition(""))
```

Comment:
It's great that an empty string is being tested here, but could you also test the cases of no parameter being passed (`rpartition()`), and passing in some non-string types?

It might also be worth testing whether it accepts an object that implements `__str__()`, to be thorough.

Also, you should test the case where `st = ""` as well.